lib/sqlalchemy/sql/selectable.py

```diff
@@ -4347,24 +4375,24 @@ def _scalar_type(self) -> TypeEngine[Any]:
     def self_group(
         self, against: Optional[OperatorType] = None
     ) -> GroupedElement:
-        return SelectStatementGrouping(self)
+        return SelectStatementGrouping(self)  # type: ignore
 
     def is_derived_from(self, fromclause: Optional[FromClause]) -> bool:
         for s in self.selects:
             if s.is_derived_from(fromclause):
                 return True
         return False
 
-    def set_label_style(self, style):
+    def set_label_style(self, style: SelectLabelStyle) -> CompoundSelect:
         if self._label_style is not style:
             self = self._generate()
             select_0 = self.selects[0].set_label_style(style)
             self.selects = [select_0] + self.selects[1:]
 
         return self
 
-    def _ensure_disambiguated_names(self):
-        new_select = self.selects[0]._ensure_disambiguated_names()
+    def _ensure_disambiguated_names(self) -> CompoundSelect:
```

Comment:
**mike bayer** (zzzeek) wrote:

remove

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4325